components/org.wso2.carbon.identity.api.server.organization.management/org.wso2.carbon.identity.api.server.organization.management.v1/src/main/java/org/wso2/carbon/identity/api/server/organization/management/v1/service/OrganizationManagementService.java

```diff
@@ -915,6 +939,59 @@ private OrganizationMetadata getOrganizationMetadataResponse(Organization organi
         return organizationMetadata;
     }
 
+    private MetaAttributesResponse getMetaAttributesResponse(Integer limit, String after, String before, String filter,
+                                                             List<String> metaAttributes, boolean recursive)
+            throws OrganizationManagementServerException {
+
+        MetaAttributesResponse metaAttributesResponse = new MetaAttributesResponse();
+
+        if (limit != 0 && CollectionUtils.isNotEmpty(metaAttributes)) {
+            boolean hasMoreItems = metaAttributes.size() > limit;
+            boolean needsReverse = StringUtils.isNotBlank(before);
+            boolean isFirstPage = (StringUtils.isBlank(before) && StringUtils.isBlank(after)) ||
+                    (StringUtils.isNotBlank(before) && !hasMoreItems);
+            boolean isLastPage = !hasMoreItems && (StringUtils.isNotBlank(after) || StringUtils.isBlank(before));
+
+            String url = "/meta-attributes" + "?limit=" + limit + "&recursive=" + recursive;
+            if (StringUtils.isNotBlank(filter)) {
+                try {
+                    url += "&filter=" + URLEncoder.encode(filter, StandardCharsets.UTF_8.name());
+                } catch (UnsupportedEncodingException e) {
+                    throw new OrganizationManagementServerException(
+                            ERROR_CODE_ERROR_BUILDING_PAGINATED_RESPONSE_URL.getMessage(),
+                            ERROR_CODE_ERROR_BUILDING_PAGINATED_RESPONSE_URL.getDescription(),
+                            ERROR_CODE_ERROR_BUILDING_PAGINATED_RESPONSE_URL.getCode(), e);
+                }
+            }
+            if (hasMoreItems) {
+                metaAttributes.remove(metaAttributes.size() - 1);
+            }
+            if (needsReverse) {
+                Collections.reverse(metaAttributes);
+            }
+            if (!isFirstPage) {
+                String encodedString = Base64.getEncoder().encodeToString(metaAttributes.get(0)
+                        .getBytes(StandardCharsets.UTF_8));
+                Link link = new Link();
+                link.setHref(URI.create(OrganizationManagementEndpointUtil.buildURIForPagination(url) + "&before="
+                        + encodedString));
+                link.setRel("previous");
+                metaAttributesResponse.addLinksItem(link);
+            }
+            if (!isLastPage) {
+                String encodedString = Base64.getEncoder().encodeToString(metaAttributes.get(metaAttributes.size() - 1)
+                        .getBytes(StandardCharsets.UTF_8));
+                Link link = new Link();
+                link.setHref(URI.create(OrganizationManagementEndpointUtil.buildURIForPagination(url) + "&after="
+                        + encodedString));
+                link.setRel("next");
```

Comment:
next, after , previous, before keywords now have more than 1 usage in this class. So better to move them to constants and reuse

Reply:
This is now updated.